### PR TITLE
Update TM performance test to use hotgym data, implement test reliability fix, and progress bar

### LINKED ISSUE #2971: Update TM performance test to use hotgym data
`temporal_memory_performance_test.py` should use hotgym data instead of fake data so that it is more representative of real results.

### PR DESCRIPTION
**Pull request by SaganBolliger**
Fixes #2971. Implements changes suggested in #2991 (does not close until changes are confirmed to fix the issue).

@chetan51 Ready for review.

### CODE REVIEW

File: tests/integration/nupic/algorithms/temporal_memory_performance_test.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env python
 # ----------------------------------------------------------------------
 # Numenta Platform for Intelligent Computing (NuPIC)
-# Copyright (C) 2014, Numenta, Inc.  Unless you have an agreement
```
Comment (chetan51) [inline tests/integration/nupic/algorithms/temporal_memory_performance_test.py]: Should be 2014-2016

**Comment by chetan51** [comment]
Done with review.

**Comment by SaganBolliger** [comment]
Investigating AV failure with compatibility test. See #2980.

**Comment by SaganBolliger** [comment]
Compatibility test does not pass. See issue #2986. May be related to performance difference.

**Comment by SaganBolliger** [comment]
I've changed the test to run the first 2000 records instead of all of hotgym since the latter was causing timeouts in CI.

**Comment by chetan51** [comment]
@rhyolight Can you restart the AppVeyor build?

**Comment by rhyolight** [comment]
Build is now queued.

**Comment by SaganBolliger** [comment]
@chetan51 @scottpurdy CI has now passed, but see my [comment here](https://github.com/numenta/nupic/issues/2991#issuecomment-179554112). Should we merge?

**Comment by chetan51** [comment]
@scottpurdy What do you think?

**Comment by SaganBolliger** [comment]
Looks like the AV build was restarted and then cancelled. Not sure why?

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**